tap_google_sheets/transform.py

```diff
@@ -0,0 +1,231 @@
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row_num, col_type))
+        return col_val
+    else:
+        return str(value)
+
+# transform boolean values in the sheet
+def transform_sheet_boolean_data(value, sheet_title, col_name, col_letter, row_num, col_type, row):
+    if isinstance(value, bool):
+        return value
+    elif isinstance(value, str):
+        if value.lower() in ('true', 't', 'yes', 'y'):
+            col_val = True
+        elif value.lower() in ('false', 'f', 'no', 'n'):
+            col_val = False
+        else:
+            col_val = str(value)
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row, col_type))
+        return col_val
+    elif isinstance(value, int):
+        if value in (1, -1):
+            col_val = True
+        elif value == 0:
+            col_val = False
+        else:
+            col_val = str(value)
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row, col_type))
+        return col_val
+
+# transform decimal values in the sheet
+def transform_sheet_decimal_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+    # Determine float decimal digits
+    decimal_digits = str(value)[::-1].find('.')
+    if decimal_digits > 15:
+        try:
+            # ROUND to multipleOf: 1e-15
+            col_val = float(round(value, 15))
+        except ValueError:
+            col_val = str(value)
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row_num, col_type))
+        return col_val
+    else: # decimal_digits <= 15, no rounding
+        try:
+            col_val = float(value)
+        except ValueError:
+            col_val = str(value)
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR: SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row_num, col_type))
+        return col_val
+
+# transform number values in the sheet
+def transform_sheet_number_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+    if isinstance(value, int):
+        return int(value)
+    elif isinstance(value, float):
+        return transform_sheet_decimal_data(value, sheet_title, col_name, col_letter, row_num, col_type)
+    else:
+        LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR: SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row_num, col_type))
+        return str(value)
+
+# return transformed column the values based on the datatype
+def get_column_value(value, sheet_title, col_name, col_letter, row_num, col_type, row):
```

Comment:
I don't see row_num argument not being used. If not used why do we need the argument being passed

Reply:
The **row_num** is being used in this block of code.